pkg/hub/submarineraddonagent/agent.go

```diff
@@ -0,0 +1,189 @@
+package submarineraddonagent
+
+import (
+	"crypto/x509"
+	"encoding/pem"
+	"fmt"
+	"path/filepath"
+	"strings"
+
+	"github.com/open-cluster-management/addon-framework/pkg/agent"
+	addonapiv1alpha1 "github.com/open-cluster-management/api/addon/v1alpha1"
+	clusterv1 "github.com/open-cluster-management/api/cluster/v1"
+	"github.com/open-cluster-management/submariner-addon/pkg/hub/submarineraddonagent/bindata"
+
+	"github.com/openshift/library-go/pkg/assets"
+	"github.com/openshift/library-go/pkg/operator/events"
+	"github.com/openshift/library-go/pkg/operator/resource/resourceapply"
+	operatorhelpers "github.com/openshift/library-go/pkg/operator/v1helpers"
+
+	certificatesv1 "k8s.io/api/certificates/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/serializer"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/klog/v2"
+)
+
+var (
+	genericScheme = runtime.NewScheme()
+	genericCodecs = serializer.NewCodecFactory(genericScheme)
+	genericCodec  = genericCodecs.UniversalDeserializer()
+)
+
+func init() {
+	scheme.AddToScheme(genericScheme)
+}
+
+const (
+	addOnName                    = "submariner-addon"
+	defaultInstallationNamespace = "open-cluster-management-agent-addon"
+)
+
+var agentDeploymentFiles = []string{
+	"pkg/hub/submarineraddonagent/manifests/clusterrole.yaml",
+	"pkg/hub/submarineraddonagent/manifests/clusterrolebinding.yaml",
+	"pkg/hub/submarineraddonagent/manifests/deployment.yaml",
+	"pkg/hub/submarineraddonagent/manifests/role.yaml",
+	"pkg/hub/submarineraddonagent/manifests/rolebinding.yaml",
+	"pkg/hub/submarineraddonagent/manifests/serviceaccount.yaml",
+}
+
+var agentHubPermissionFiles = []string{
+	"pkg/hub/submarineraddonagent/manifests/hub_role.yaml",
+	"pkg/hub/submarineraddonagent/manifests/hub_rolebinding.yaml",
+}
+
+// addOnAgent monitors the Submariner agent status and configure Submariner cluster environment on the managed cluster
+type addOnAgent struct {
+	agentName  string
+	kubeClient kubernetes.Interface
+	recorder   events.Recorder
+}
+
+// NewAddOnAgent returns an instance of addOnAgent
+func NewAddOnAgent(agentName string, kubeClient kubernetes.Interface, recorder events.Recorder) *addOnAgent {
+	return &addOnAgent{
+		agentName:  agentName,
+		kubeClient: kubeClient,
+		recorder:   recorder,
+	}
+}
+
+// Manifests generates manifestworks to deploy the submariner-addon agent on the managed cluster
+func (a *addOnAgent) Manifests(cluster *clusterv1.ManagedCluster, addon *addonapiv1alpha1.ManagedClusterAddOn) ([]runtime.Object, error) {
```

Comment:
I think you would also need to create submariner sa here?

Reply:
we don't need a submariner sa here, it will be created by hub controller